dpnp/backend/src/dpnp_iterator.hpp

```diff
@@ -485,35 +477,30 @@ class DPNPC_id final
         {
             assert(output_global_id < output_size);
 
-            // use thread private storage
-            size_type* sycl_output_xyz_thread = sycl_output_xyz + (output_global_id * output_shape_size);
-
-            get_xyz_by_id_inkernel(output_global_id, output_shape_strides, output_shape_size, sycl_output_xyz_thread);
-
             for (size_t iit = 0, oit = 0; iit < input_shape_size; ++iit)
             {
                 if (std::find(axes.begin(), axes.end(), iit) == axes.end())
                 {
-                    input_global_id += (sycl_output_xyz_thread[oit] * input_shape_strides[iit]);
+                    const size_type output_xyz_id = get_xyz_id_by_id_inkernel(output_global_id, output_shape_strides,
```

Comment:
Is it used for broadcast?

Reply:
It's used for both reducing and broadcasting.